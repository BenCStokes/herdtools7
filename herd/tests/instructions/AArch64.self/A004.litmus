AArch64 A004
(* Herd gives incorrect result at the moment,
 * because instruction B L02 on P1 is
 * referenced to by two labels. *)
"PodWW Irfe PodRR Ifre"
Generator=diyone7 (version 7.56+02~dev)
Com=Irf Ifr
Orig=PodWW Irfe PodRR Ifre
{
0:X0=NOP; 0:X1=P1:Lself00; 0:X2=P1:Lself01;
}
 P0          | P1        ;
 STR W0,[X1] | Lself01:  ;
 STR W0,[X2] | B L00     ;
             | MOV W0,#2 ;
             | B L01     ;
             | L00:      ;
             | MOV W0,#1 ;
             | L01:      ;
             | Lself00:  ;
             | B L02     ;
             | MOV W1,#2 ;
             | B L03     ;
             | L02:      ;
             | MOV W1,#1 ;
             | L03:      ;
exists (1:X0=2 /\ 1:X1=1)
