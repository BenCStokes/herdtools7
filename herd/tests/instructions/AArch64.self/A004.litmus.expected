Test A004 Allowed
States 4
1:X0=1; 1:X1=1;
1:X0=1; 1:X1=2;
1:X0=2; 1:X1=1;
1:X0=2; 1:X1=2;
Ok
Witnesses
Positive: 1 Negative: 3
Condition exists (1:X0=2 /\ 1:X1=1)
Observation A004 Sometimes 1 3
Hash=a4610bdbbb2405e7c82c9a4b56acbae1

